numpy/core/src/multiarray/compiled_base.c

```diff
@@ -619,6 +623,14 @@ arr_interp(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwdict)
         }
 
         NPY_END_THREADS;
+
+        if (found_nan) {
+            // 2019-07-13, NumPy 1.18.0
+            if (DEPRECATE("`np.interp` found NaN in xp. This will raise an "
+                    "error starting in NumPy 1.19.0.") < 0) {
+                goto fail;
+            }
```

Comment:
Instead of deprecating this, should we just give a runtime warning through the typical errstate mechanism?

Reply:
We probably have to discuss it briefly. This touches the `xp`, so the places where the values are defined. Not the `x` values where we want interpolation to occur. To me that seems like a deprecation? (Unless we decide to just make a big warning in the documentation that NaNs are not sortable and thus its your fault if you pass them in)

I agree that it may be right to check errorflags for the any other result, but I cannot wrap my head into how to handle NaNs in `xp` reasonably.